pandas/io/pytables.py

```diff
@@ -387,7 +387,10 @@ def read_hdf(
         if key is None:
             groups = store.groups()
             if len(groups) == 0:
-                raise ValueError("No dataset in HDF5 file.")
+                raise ValueError(
+                    "Dataset that is either compatible with Pandas data types "
+                    "or is a table could not be found in HDF5 file."
+                )
```

Comment:
can u add a file with an incompatible dataset (eg create in h5py)

Reply:
Working on it...

Reply:
OK, added a file with an incompatible `dataset, pandas/tests/io/data/legacy_hdf/incompatible_dataset.h5` and a test